### DISCUSSION TITLE
How to set Referer HTTP header?

### DISCUSSION THREAD

**Discussion by nicolas-raoul** [osmdroid/osmdroid#1961] [Q&A] [answered]
Wikimedia wants me to set a `Referer` to use their tiles (I am developing an official Wikimedia app).

What is the recommended way to set that HTTP header?

I see that `mAdditionalHttpRequestProperties` is `private final` with no setter: https://github.com/osmdroid/osmdroid/blob/be33d9814e99d6289dd80e0ffdcb77b87cb333f9/osmdroid-android/src/main/java/org/osmdroid/config/DefaultConfigurationProvider.java#L46

I see a similar question at the old https://raw.githubusercontent.com/osmdroid/osmdroid/master/issues/osmdroid_issue249.json and https://raw.githubusercontent.com/osmdroid/osmdroid/master/issues/osmdroid_issue432.json but the outcome is not clear to me.

Thanks, keep up the great work!

**Comment by mnalis** [comment]
While there is not a classical setter, there seems to be `load` (as well as `save`) method, which perhaps might be used for setting `mAdditionalHttpRequestProperties` ?

https://github.com/osmdroid/osmdroid/blob/be33d9814e99d6289dd80e0ffdcb77b87cb333f9/osmdroid-android/src/main/java/org/osmdroid/config/DefaultConfigurationProvider.java#L361

(I'm no android nor Java/Kotlin programmer, though, so no much ideas _how_ to actually do it)

**Comment by spyhunter99** [comment, marked as answer]
i think you're looking for

    Configuration.getInstance().getAdditionalHttpRequestProperties().put("myHeader", "value");

**Reply by mnalis** [reply]
thanks @spyhunter99 for the hint; I've forwarded that information to https://github.com/commons-app/apps-android-commons/issues/5374#issuecomment-1873511233 (from which this issue originated)
